src/Assets/MagneticField/FieldLoader.js

```diff
@@ -33,8 +62,35 @@ class FieldLoader {
      */
     _RenderData(data, scene) {
         let field_instance = new MagneticField(data);
-        scene.AddAsset(field_instance);
+        this.field_instances.push(field_instance);
     }
 }
+class LineManager{
+    constructor(MagneticFieldInstances, scene) {
+        this.scene = scene;
+        this.lines = MagneticFieldInstances;
+        scene.RegisterTimeUpdateListener(this.listener.bind(this)) //ask about this 
+    }
+    async listener(CurrDate){
+        let result = ToAPIDate(CurrDate);
+        // console.log(this.lines[0]['_data']['fieldlines']['frame']['source_map_obstime']['value']);
+        let right_url = await Helios.get_field_lines(result);
+        for (let i = 0; i < this.lines.length; i++) {
+            // let date = this.lines[i].getDate();
+            let date = this.lines[i]['_data']['fieldlines']['frame']['source_map_obstime']['value'];
+            // let real_date = ToAPIDate(date);
+            // console.log(date);
+            let url = await Helios.get_field_lines(date);
+            // console.log(url,right_url);
+            console.log(right_url);
+            console.log(url);
+            if (right_url === url){
+                console.log("sameeeeeeeee");
+                this.scene.AddAsset(this.lines[i]);
```

Comment:
Instead of adding the instance of MagneticField from `this.lines[i]` I think this `LineManager` instance should be the one added to the scene. Only add it once in the constructor.

`LineManager` should implement the interface described [here](https://github.com/Helioviewer-Project/helios/tree/assets/src/Assets)

Instead of calling `RegisterTimeUpdateListener`, rename listener to `SetTime` and it will be called automatically instead of needing to register a listener.

`LineManager` should have an instance of [Group](https://threejs.org/docs/#api/en/objects/Group) where you can show/hide the correct set of field lines. `GetRenderableModel()` will return this group instance.